### PR TITLE
New tests for receiving a window size before sending SYN

### PR DESCRIPTION
**Pull request by avajihschiff**
Tests mentioned on ed post #570

### CODE REVIEW

File: tests/send_extra.cc [hunk 1]
```diff
@@ -565,6 +565,36 @@ int main()
...
       test.execute( HasError { true } );
     }
+    // My extra tests: Ava Jih-Schiff
+    {
+      TCPConfig cfg;
+      const Wrap32 isn( rd() );
+      cfg.isn = isn;
+
+      TCPSenderTestHarness test { "Receiving a window size before transmit I", cfg };
+      test.execute( AckReceived { Wrap32 { 0 } }.with_win( 3 ).without_push() ); // invalid ack but window is set
```
Comment (ammrat13) [inline tests/send_extra.cc:575]: This line will have an `ackno` of zero, but `ackno` should have an offset from the initial sequence number. I believe this line should create a `Wrap32` with `cfg.isn` instead.

File: tests/send_extra.cc [hunk 2]
```diff
@@ -565,6 +565,36 @@ int main()
...
+      test.execute( AckReceived { Wrap32 { isn + 4 } }.with_win( 1 ) );
+      test.execute( ExpectMessage {}.with_no_flags().with_fin( true ).with_payload_size( 0 ).with_seqno( isn + 4) ); 
+    }
+    {
+      TCPConfig cfg;
+      const Wrap32 isn( rd() );
+      cfg.isn = isn;
+
+      TCPSenderTestHarness test { "Receiving a window size before transmit II", cfg };
+      test.execute( AckReceived { Wrap32 { 0 } }.with_win( 5 ).without_push() ); // invalid ack but window is set
```
Comment (ammrat13) [inline tests/send_extra.cc:592]: Same as above

**Review by ammrat13 — CHANGES_REQUESTED** [review]

File: tests/send_extra.cc (hunk 2 above)
Comment (avajihschiff) [reply tests/send_extra.cc]: Fixed!

**Review by ammrat13 — APPROVED** [review]

**Comment by keithw** [comment]
Thanks! We'll take the first one upstream. The second one (honoring the winsize in a ReceiverMessage that has an invalid ackno) probably can't be a normative requirement -- it's okay to ignore these messages completely.

**Pull request closed without merging**